File: android/app/src/main/java/cn/edu/jmu/openjmu/plugin/SecureFlagPlugin.java
package cn.edu.jmu.openjmu.plugin;

import android.app.Activity;
import android.view.WindowManager;

import androidx.annotation.NonNull;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.PluginRegistry;

public class SecureFlagPlugin implements FlutterPlugin, ActivityAware,
        MethodChannel.MethodCallHandler {
    private static final String channelName = "cn.edu.jmu.openjmu/setFlagSecure";
    private Activity activity;

    public static void registerWith(PluginRegistry.Registrar registrar) {
        final SecureFlagPlugin plugin = new SecureFlagPlugin();
        plugin.onAttachedToEngine(registrar.messenger(), registrar.activity());
    }

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
        onAttachedToEngine(binding.getBinaryMessenger(), null);
    }

    private void onAttachedToEngine(BinaryMessenger messenger, Activity activity) {
        MethodChannel methodChannel = new MethodChannel(messenger, channelName);
        methodChannel.setMethodCallHandler(this);
        if (activity != null) this.activity = activity;
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
        activity = null;
    }

    @Override
    public void onMethodCall(MethodCall call, @NonNull MethodChannel.Result result) {
        if (call.method.equals("enable")) {
            activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
            result.success("Flag Secured Success.");
        } else if (call.method.equals("disable")) {
            activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
            result.success("Flag InSecured Success.");
        }
    }

    @Override
    public void onAttachedToActivity(ActivityPluginBinding binding) {
        this.activity = binding.getActivity();
    }

    @Override
    public void onDetachedFromActivityForConfigChanges() {
        this.activity = null;
    }

    @Override
    public void onReattachedToActivityForConfigChanges(ActivityPluginBinding binding) {
        this.activity = binding.getActivity();
    }

    @Override
    public void onDetachedFromActivity() {
        this.activity = null;
    }
}

File: android/app/src/main/kotlin/cn/edu/jmu/openjmu/plugin/SecureFlagPlugin.kt
package cn.edu.jmu.openjmu.plugin

import android.app.Activity
import android.view.WindowManager
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.*
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

class SecureFlagPlugin : FlutterPlugin, ActivityAware, MethodCallHandler {
    private var activity: Activity? = null
    override fun onAttachedToEngine(binding: FlutterPluginBinding) {
        val methodChannel = MethodChannel(binding.binaryMessenger, channelName)
        methodChannel.setMethodCallHandler(this)
    }

    override fun onDetachedFromEngine(binding: FlutterPluginBinding) {
        activity = null
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        if (call.method == "enable") {
            activity!!.window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
            result.success("Flag Secured Success.")
        } else if (call.method == "disable") {
            activity!!.window.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
            result.success("Flag InSecured Success.")
        }
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivityForConfigChanges() {
        activity = null
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivity() {
        activity = null
    }

    companion object {
        private const val channelName = "cn.edu.jmu.openjmu/setFlagSecure"
    }
}